etna/transforms/timestamp/holiday.py

```diff
@@ -60,19 +95,42 @@ def _get_column_name(self) -> str:
             return self.__repr__()
 
     def _fit(self, df: pd.DataFrame) -> "HolidayTransform":
+        """Fit the transform.
+
+        Parameters
+        ----------
+        df:
+            Dataset to fit the transform on.
+
+        Returns
+        -------
+        :
+            The fitted transform instance.
         """
-        Fit HolidayTransform with data from df. Does nothing in this case.
+        return self
+
+    def fit(self, ts: TSDataset):
+        """Fit the transform.
 
         Parameters
         ----------
-        df: pd.DataFrame
-            value series with index column in timestamp format
+        ts:
+            Dataset to fit the transform on.
+
+        Returns
+        -------
+        :
+            The fitted transform instance.
         """
+        df = ts.to_pandas(flatten=False, features=self.required_features)
+        self._fit(df=df)
+        self.freq = ts.freq
         return self
 
     def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
         """
         Transform data from df with HolidayTransform and generate a column of holidays flags or its titles.
```

Comment:
It isn't a good docstring according to numpydoc format. Short description should be one line and then goes extended description which could consist of many lines. We could keep only `Transform the data` here I guess.